llvm/lib/Transforms/Utils/BuildLibCalls.cpp

```diff
@@ -512,6 +512,7 @@ bool llvm::inferNonMandatoryLibFuncAttrs(Function &F,
     Changed |= setAllocKind(F, AllocFnKind::Alloc | AllocFnKind::Uninitialized | AllocFnKind::Aligned);
     [[fallthrough]];
   case LibFunc_valloc:
+  case LibFunc_pvalloc:
```

Comment:
Inferring allocsize is incorrect here, because pvalloc will round up the passed size.

Reply:
Good point - removed the AllocSize attr.